java/gateway-core/src/main/java/com/avaulta/gateway/rules/JsonSchemaFilterUtils.java

```diff
@@ -311,6 +402,48 @@ public boolean hasType() {
         public boolean isComplex() {
             return isObject() || isArray();
         }
+
+        @JsonIgnore
+        public boolean hasIf() {
+            return this._if != null;
+        }
+
+        @JsonIgnore
+        public boolean hasElse() {
+            return this._else != null;
+        }
+
+        @JsonIgnore
+        public boolean hasThen() {
+            return this._then != null;
+        }
+
+        @JsonIgnore
+        public boolean hasConstant() {return this.constant != null; }
+
+        @JsonIgnore
+        public boolean hashOneOf() {return this.oneOf != null && !this.oneOf.isEmpty();}
+    }
+
+    /**
+     * For building if-else-then conditions in json schema
+     * See <a href="https://json-schema.org/understanding-json-schema/reference/conditionals.html#if-then-else">...</a>
+     */
+    @SuperBuilder(toBuilder = true)
+    @NoArgsConstructor
+    public static class ConditionJsonSchema extends JsonSchemaFilter {
     }
 
+    @SuperBuilder(toBuilder = true)
+    @NoArgsConstructor
+    public static class ThenJsonSchema extends JsonSchemaFilter {
+    }
+
+    private static class NotMatchedConstant {
```

Comment:
Not clean, but I need something *defined* to propagate in the object structure to differentiate between an object with json content or the condition not matched.